### ISSUE TITLE
Build error with "nlohmann/json.hpp: No such file or directory" in spack (unit tests build, with --test=root option)

### ISSUE THREAD

**Issue by wkensuke** [root-project/root#14769]
### Check duplicate issues.

- [X] Checked for duplicates

### Description

Hi, I use Spack to evaluate Fujitsu compilers for Fugaku and FX1000.
When I tried to install this application on spack with the "--test=root" option(unit tests), I got the following error:

```
/fefs/spack2/tmp/spack-stage/spack-stage-root-6.30.02-ifueiwdgknhz2zyg3olctrljdchct3mv/spack-src/tree/dataframe/test/dataframe_datasetspec.cxx:15:10: fatal error: nlohmann/json.hpp: No such file or directory
   15 | #include <nlohmann/json.hpp>
      |          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [tree/dataframe/test/CMakeFiles/dataframe_datasetspec.dir/build.make:79: tree/dataframe/test/CMakeFiles/dataframe_datasetspec.dir/dataframe_datasetspec.cxx.o] Error 1
make[2]: Leaving directory '/fefs/spack2/tmp/spack-stage/spack-stage-root-6.30.02-ifueiwdgknhz2zyg3olctrljdchct3mv/spack-build-ifueiwd'
make[1]: *** [CMakeFiles/Makefile2:51492: tree/dataframe/test/CMakeFiles/dataframe_datasetspec.dir/all] Error 2
make[1]: Leaving directory '/fefs/spack2/tmp/spack-stage/spack-stage-root-6.30.02-ifueiwdgknhz2zyg3olctrljdchct3mv/spack-build-ifueiwd'
make: *** [Makefile:169: all] Error 2
```

Attached build log.
[spack-build-out_root.txt](https://github.com/root-project/root/files/14339928/spack-build-out_root.txt)

### Reproducer

``` 
$ spack install --test=root root@6.30.02%gcc@12.2.0
```

### ROOT version

[v6-30-02](https://github.com/root-project/root/releases/tag/v6-30-02)

### Installation method

Spack package manager

### Operating system

Linux(RHEL 8.8,aarch64)

### Additional context

There is another problem when using Spack. 
Please check the issue of spack.
[Installation issue: root@6.30.02 fails to build with --test=root option #42720](https://github.com/spack/spack/issues/42720)

**Comment by ferdymercury** [comment]
Please try with master branch, or with 6.30.04, this seems a duplicate of:
https://github.com/root-project/root/issues/14188
which was recently fixed.

**Comment by vvolkl** [comment]
Or add https://github.com/root-project/root/pull/14192/files.patch as a patch.

**Comment by ferdymercury** [comment]
@linev might know.
Maybe it's not until 6.30.06 that it was fixed. Please compare the CMakeLists of master vs yours and try editing it by hand.

**Comment by linev** [comment]
As I can see from the full log, `'-Dbuiltin_nlohmannjson:BOOL=OFF'` is set there.
Means external `nlohmann/json.hpp` will be used when building ROOT.

In such case all tests also have to use external includes. I will try to provide fix.

**Comment by linev** [comment]
Please try https://github.com/root-project/root/pull/14784

**Comment by wkensuke** [comment]
@linev @vvolkl 

Thank you.

> Please try https://github.com/root-project/root/pull/14784
I built with this patch and was able to complete successfully.
